src/libraries/Enableable.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {RolesConsumer} from "src/modules/ROLES/OlympusRoles.sol";
+
+/// @title  Enableable
+/// @notice This contract is designed to be inherited by contracts that need to be enabled or disabled. It replaces the inconsistent usage of `active` and `locallyActive` state variables across the codebase.
+/// @dev    A contract that inherits from this contract should use the `whenEnabled` and `whenDisabled` modifiers to gate access to certain functions.
+///
+///         If custom logic and/or parameters are needed for the enable/disable functions, the inheriting contract can override the `_enable()` and `_disable()` functions. For example, `enable()` could be called with initialisation data that is decoded, validated and assigned in `_enable()`.
+abstract contract Enableable is RolesConsumer {
+    // ===== STATE VARIABLES ===== //
+
+    bool public isEnabled;
+    bytes32 public constant ROLE = "emergency_shutdown";
```

Comment:
imo there should be a `RoleDefinitions.sol` somewhere with all of these in the one place, which this file imports.

Can be constants in the global namespace:

`RoleDefinitions.sol`:
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

/// @notice Allows shutting down the protocol and other actions in an emergency
bytes32 constant ROLE = "emergency_shutdown";

...
```

Reply:
The problem is that so many policies have defined their own roles ("cooler_overseer", "custodian", etc) and there's no standardization. I was thinking that it may be easier to define three levels for the standard roles ("admin", "manager", "emergency") and then allow for custom/nuanced roles. The standard roles can be added into RoleDefinitions.sol.